src/window.rs

```diff
@@ -92,23 +92,19 @@ impl WindowId {
     pub const fn dummy() -> Self {
         WindowId(platform_impl::WindowId::dummy())
     }
-}
 
-impl fmt::Debug for WindowId {
-    fn fmt(&self, fmtr: &mut fmt::Formatter<'_>) -> fmt::Result {
-        self.0.fmt(fmtr)
+    pub fn into_raw(self) -> u64 {
```

Comment:
I'm still not entirely sure where these are useful, maybe we could add that as documentation? But not a blocker for this PR, just a nicety

Reply:
From what I understood in the comments above they are used to have an identifier to be able to communicate with outside of Winit, e.g. through environment variables and other crates.
While this doesn't really convince me to have this API in place, I don't properly understand the use-cases either.

But I agree we should leave documentation here, but I will let others do this who actually understand the use-cases.

Reply:
@madsmtm One of the main use cases for `WindowId` in general is to:
- Call `window.id()` to get an `WindowId`
- Store it for later use
- And then later pass it back into Winit

This PR allows you to roundtrip a `WindowId` into a built-in type and back into a `WindowId`. Which is useful if the place you want to store the `WindowId` doesn't otherwise depend on Winit: it can then just store a u64 (which it treats as opaque), which gets converted back into a `WindowId` by an outer wrapper layer that does know about Winit.

Basically it allows for separation of concerns.

Reply:
This is a concept I understand.
But I'm interested in an actual use-case that this facilitates and some reason why Winit has to help out there.

The concept could be argued for any type really, e.g. it could be said about `DeviceId`, `FingerId`, `CustomCursor` or `MonitorHandle`. I hope my point of view is understandable!

I apologize for any naivety on my part here, definitely wish I was more aware of the ecosystem Winit is used and consumed in.

Reply:
I've added a little bit of documentation to these types now, feel free to say so if you believe it should say something else.